### PR TITLE
follow-up: removal of the ratelimit ir

### PR DESCRIPTION
**Pull request by shawnh2**
**What type of PR is this?**

this PR will remove all logic related to ratelimit ir.

**What this PR does / why we need it**:

we don't need it anymore since the ratelimit have been switched to XDS mode, and the ratelimit can be created by config statically.

**Which issue(s) this PR fixes**:

related to #1416

### CODE REVIEW

File: internal/cmd/server.go [hunk 1]
```diff
@@ -175,15 +173,19 @@ func setupRunners(cfg *config.Server) error {
...
 
-	// Start the global rateLimit runner if it has been enabled through the config
+	// Start the global rateLimit if it has been enabled through the config
 	if cfg.EnvoyGateway.RateLimit != nil {
-		// Start the Global RateLimit Runner
-		// It subscribes to the xds Resources and translates it to Envoy Ratelimit Service
-		// infrastructure and configuration.
+		// Create the global ratelimit infra statically.
+		mgr := infraRunner.Manager()
+		if err := mgr.CreateOrUpdateRateLimitInfra(ctx); err != nil {
```
Comment (qicz) [inline internal/cmd/server.go:180]: prefer to  move to `runner`

Comment (arkodg) [reply internal/cmd/server.go:180]: +1 , prefer if this lives inside the infra-manager runner
Also you can block on `<-ctx.Done()` in https://github.com/envoyproxy/gateway/blob/56edabaac958d34254028b8fb9f0f68ed3c1ac9d/internal/infrastructure/runner/runner.go#L51
and delete the RL service if it was created before

Comment (shawnh2) [reply internal/cmd/server.go:180]: sounds really great

**Review by qicz — DISMISSED** [review]
LGTM

**Review by arkodg — APPROVED** [review]
LGTM, thanks for the clean up

**Pull request merged**